sdk/core/System.ClientModel/tests/TestFramework/Logging/TestLogger.cs

```diff
@@ -0,0 +1,57 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.ClientModel.Tests.TestFramework;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Extensions.Logging;
+
+namespace ClientModel.Tests;
+
+public class TestLogger : ILogger
```

Comment:
Why is an ILogger needed for testing?

Reply:
Since ILogger is just an abstraction for allowing customers to log things wherever they want, this is a skeleton implementation to use for the tests. It puts all the written logs into a collection and does some light formatting to make it easy to check formatting / log state. It's a really similar approach to TestClientEventListener